deepinv/training/run_logger.py

```diff
@@ -0,0 +1,499 @@
+    def log_losses(
+        self,
+        losses: dict[str, float],
+        step: int,
+        epoch: int | None = None,
+        phase: str = "train",
+    ) -> None:
+
+        # Human readable logging
+        loss_str = "| ".join([f"{name}: {value:.6f}" for name, value in losses.items()])
+        self.stdout_logger.info(
+            f"{phase} - epoch: {epoch} | step: {step} | losses: {loss_str}"
+        )
+
+        # JSON logging
+        log_file = self.loss_dir / "losses.json"
+
+        if log_file.exists():
+            try:
+                with open(log_file, "r") as f:
+                    all_logs = json.load(f)
+            except json.JSONDecodeError:
+                all_logs = {}
+        else:
+            all_logs = {}
+
+        if phase not in all_logs:
+            all_logs[phase] = []
+
+        entry = {
+            "epoch": epoch,
+            "step": step,
+            "losses": {name: float(value) for name, value in losses.items()},
+        }
+
+        # Add the new entry to the list
+        all_logs[phase].append(entry)
+
+        with open(log_file, "w") as f:
+            json.dump(all_logs, f, indent=2)
+
+    def log_metrics(
+        self,
+        metrics: dict[str, float],
+        step: int,
+        epoch: int | None = None,
+        phase: str = "train",
+    ) -> None:
+        # Human readable logging
+        metric_str = "| ".join(
+            [f"{name}: {value:.6f}" for name, value in metrics.items()]
+        )
+        self.stdout_logger.info(
+            f"{phase} - epoch: {epoch} | step: {step} | metrics: {metric_str}"
+        )
+
+        # JSON logging
+        log_file = self.metrics_dir / "metrics.json"
+
+        if log_file.exists():
+            try:
+                with open(log_file, "r") as f:
+                    all_logs = json.load(f)
+            except json.JSONDecodeError:
+                all_logs = {}
+        else:
+            all_logs = {}
+
+        if phase not in all_logs:
+            all_logs[phase] = []
+
+        entry = {
+            "epoch": epoch,
+            "step": step,
+            "metrics": {name: float(value) for name, value in metrics.items()},
+        }
+
+        all_logs[phase].append(entry)
+
+        with open(log_file, "w") as f:
+            json.dump(all_logs, f, indent=2)
+
+    def log_images(
+        self,
+        images: dict[str, torch.Tensor],
+        epoch: int = 0,
+        step: int | None = None,
+        phase: str = "train",
+    ) -> None:
+        dir_path = self.images_dir / phase / f"epoch_{epoch}"
+        if step is not None:
+            dir_path = dir_path / f"step_{step}"
+        os.makedirs(dir_path, exist_ok=True)
+
+        for k, (name, img) in enumerate(images.items()):
+            for i in range(img.size(0)):
+                img_name = f"{dir_path}/{name}_{k}_{self.img_counter + i}.png"
+                save_image(img[i], img_name)
+
+    def load_from_checkpoint(self, checkpoint_dict: dict[str, Any]):
```

Comment:
Given the name of this method, I would expect it to load the checkpoint: model weights. optimizer weights... Is this not true?